### DISCUSSION TITLE
Relax the requirement on `-sources` and `-javadoc` for Maven Central rules

### DISCUSSION THREAD

**Discussion by gastaldi** [jreleaser/jreleaser#1750] [Q&A]
There are some JARs that contain only resources (and some are empty), thus it makes no sense to have a `-sources` and a `-javadoc` in these cases.

My use case is a bit different, so using `artifactOverrides` isn't an option here as I'm using a centralized `jreleaser.yml` which is invoked for deployment (https://github.com/quarkiverse/quarkiverse-release/blob/main/jreleaser.yml).

Would it make sense if pomchecker checked if any `.class` exists inside the JAR before requiring a `-sources` and `-javadoc`? IMHO that would be the only reason those would be needed.

**Comment by aalmiray** [comment]
Pomchecker is not responsible for this behavior, JReleaser is. 

It's possible to globally skip checking for sources and javadoc JARs when configured at the deployer level, instead of using `artifactOverrides`. 

Opening a JAR and checking its contents takes time. Also, there are JAR files that only contain resource files which I believe will still trigger the sources rule check performed by Maven Central. Thus that solution is not optional at the moment. 

This being said, I'll reach to Sonatype to find out more about the checks they perform on their side and see if they can be replicated by JReleaser.

**Reply by gastaldi** [reply]
> It's possible to globally skip checking for sources and javadoc JARs when configured at the deployer level, instead of using `artifactOverrides`.

Disabling globally may not be a good idea, since it would skip the cases where a JAR with classes should have produced its sources and javadoc correspondents.

**Reply by aalmiray** [reply]
Correct. Finding the exact checks applied to each artifact is key. For now is all of them (global) or explicit per artifact (`artifactOverrides`).

Having checks applied implicitly may be tricky but I can find out what's needed.
